packages/jest-resolve/src/utils.ts

```diff
@@ -98,6 +98,11 @@ export const resolveTestEnvironment = ({
   testEnvironment: string;
   requireResolveFunction: (moduleName: string) => string;
 }): string => {
+  // we don't want to resolve the actual `jsdom` module if `jest-environment-jsdom` is not installed, but `jsdom` package is
+  if (filePath === 'jsdom') {
+    filePath = 'jest-environment-jsdom';
+  }
```

Comment:
@SimenB: Doesn't the `prefix` argument in `resolveWithPrefix` handle this already? I'm finding on yarn that the code attempts to resolve `jest-environment-jest-environment-jsdom`.

Reply:
Okay, never mind -- `resolveWithPrefix` tries both with a prefix and without a prefix.